core/src/license.rs

```diff
@@ -25,6 +26,11 @@ use crate::request::{Request, REQ_PLAINTEXT_SIZE};
 pub(crate) const LIC_PLAINTEXT_SIZE: usize = Signature::SIZE + JubJubScalar::SIZE;
 const LIC_ENCRYPTION_SIZE: usize = LIC_PLAINTEXT_SIZE + ENCRYPTION_EXTRA_SIZE;
 
+pub enum LicenseCreator {
```

Comment:
I'd rename this enum to make it a unanimated noun. Name like "creator" suggests it is a function or a module, and it is very misleading. It could be a LicenseSource or LicenseOrigin or sth. like that.